server/user.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var childProcess = require('child_process');
+var readline = require('readline');
+var mkdirp = require('mkdirp');
+
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGHUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!Object} userInfo User Information
+ * @return {!Object{
```

Comment:
I don't think user.Error is necessary, why not just use user.SignupResponse like this?
```js
/**
 * @typedef {{
 *   success: boolean,
 *   errorMessage: {=string}
 * }}
 */
user.SignupResponse;
```